### ISSUE TITLE
Move to using MiB for defining Task memory requirements

### ISSUE THREAD

**Issue by delagoya** [ga4gh/task-execution-schemas#116]
AWS Batch, Azure Kubernetes, and GCP all seem to use MiB for memory definitions. They also use `integer` instead of `number`

Propose we move to this model for defining Task memory requirements over a `double` GB.

**Comment by geoffjentry** [comment]
I agree on using MiB. Besides the reason you state, we recently encountered some issues in Cromwell as users were making hyperoptimized WDLs and running afoul of things as they didn't understand the difference (WDL supports both units). What we wound up doing was always treating MB as MiB (and so forth up the unit chain) as for those types of users that's usually what they meant anyways, and MiB > MB so it was the safer fallback
